sdks/java/io/jdbc/src/test/java/org/apache/beam/sdk/io/jdbc/JdbcIOTest.java

```diff
@@ -490,6 +496,149 @@ public void tearDown() {
     }
   }
 
+  @Test
+  public void testWriteWithoutPreparedStatement() throws Exception {
```

Comment:
It's fine to test with the actual database but I don't think all possible datatypes are covered here. It would be good to use mocks and make sure all your prepared statement setters are exercised. (Use the coverage report as a guide)

Reply:
Done, Added tests for prepared statement caller